src/particle/src/interaction/4C_particle_interaction_dem_adhesion_surface_energy.hpp

```diff
@@ -8,111 +8,37 @@
-
-    //! calculate adhesion surface energy
-    virtual void adhesion_surface_energy(
-        const double& mean_surface_energy, double& surface_energy) const = 0;
 
-   protected:
-    //! discrete element method parameter list
-    const Teuchos::ParameterList& params_dem_;
-  };
-
-  class DEMAdhesionSurfaceEnergyConstant : public DEMAdhesionSurfaceEnergyBase
+  struct DEMAdhesionSurfaceEnergyParams
   {
-   public:
-    //! constructor
-    explicit DEMAdhesionSurfaceEnergyConstant(const Teuchos::ParameterList& params);
-
-    //! get adhesion surface energy
-    void adhesion_surface_energy(
-        const double& mean_surface_energy, double& surface_energy) const override
-    {
-      surface_energy = mean_surface_energy;
-    };
+    SurfaceEnergyDistribution type;
+    double surface_energy;
+    double distribution_standard_deviation;
+    double distribution_cutoff_factor;
   };
 
-  class DEMAdhesionSurfaceEnergyDistributionBase : public DEMAdhesionSurfaceEnergyBase
-  {
-   public:
-    //! constructor
-    explicit DEMAdhesionSurfaceEnergyDistributionBase(const Teuchos::ParameterList& params);
-
-    //! setup adhesion surface energy handler
-    void setup() override;
-
-    //! get adhesion surface energy
-    void adhesion_surface_energy(
-        const double& mean_surface_energy, double& surface_energy) const override = 0;
+  void verify_params_adhesion_surface_energy(const DEMAdhesionSurfaceEnergyParams& params);
 
-   protected:
-    //! adjust surface energy to allowed bounds
-    void adjust_surface_energy_to_allowed_bounds(
-        const double& mean_surface_energy, double& surface_energy) const;
-
-    //! variance of adhesion surface energy distribution
-    const double variance_;
-
-    //! cutoff factor of adhesion surface energy to determine minimum and maximum value
-    const double cutofffactor_;
-  };
-
-  class DEMAdhesionSurfaceEnergyDistributionNormal : public DEMAdhesionSurfaceEnergyDistributionBase
-  {
-   public:
-    //! constructor
-    explicit DEMAdhesionSurfaceEnergyDistributionNormal(const Teuchos::ParameterList& params);
-
-    //! get adhesion surface energy
-    void adhesion_surface_energy(
-        const double& mean_surface_energy, double& surface_energy) const override;
-  };
-
-  class DEMAdhesionSurfaceEnergyDistributionLogNormal
-      : public DEMAdhesionSurfaceEnergyDistributionBase
-  {
-   public:
-    //! constructor
-    explicit DEMAdhesionSurfaceEnergyDistributionLogNormal(const Teuchos::ParameterList& params);
-
-    //! get adhesion surface energy
-    void adhesion_surface_energy(
-        const double& mean_surface_energy, double& surface_energy) const override;
-  };
+  /*!
+   * Calculates the surface energy based on the given SurfaceEnergyDistribution type in params and
+   * the given mean_surface_energy value.
+   *
+   * \note the surface energy from DEMAdhesionSurfaceEnergyParams is not used because the function
+   * is also used for the ParticleWall
```

Comment:
Is there something missing? Why is the ParticleWall related to this function? Not clear to non-experts ;)

Reply:
I split the params and introduced a new adhesion_params struct. With further restructuring I will add more parameters to this struct in a future PR.